programs/cleos/CMakeLists.txt

```diff
@@ -16,9 +16,16 @@ if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../../.git)
     execute_process(
       COMMAND ${GIT_EXECUTABLE} rev-parse --short=8 HEAD
       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../.."
-      OUTPUT_VARIABLE "cleos_BUILD_VERSION"
+      OUTPUT_VARIABLE "cleos_BUILD_HASH"
       ERROR_QUIET
       OUTPUT_STRIP_TRAILING_WHITESPACE)
+    execute_process(
```

Comment:
this approach is only evaluated during cmake generation so it's not as sensitive as it should be. Compare to how it was implemented for nodeos:
```
$ ./programs/nodeos/nodeos --version
v1.8.0-rc1-501-gafa686d8c
$ ./programs/cleos/cleos version full
1.9.0-develop-afa686d8
$ echo "// spoon was here" >> ../src/programs/cleos/main.cpp 
$ ninja
...
$ ./programs/nodeos/nodeos --version
v1.8.0-rc1-501-gafa686d8c-dirty
$ ./programs/cleos/cleos version full
1.9.0-develop-afa686d8
```

I think we should consider refactoring out the more advanced appbase/nodeos code in a common manner so that it can be used by both nodeos as well as non-appbase based targets like cleos. Probably never belonged in appbase in the first place. Plus that will reduce duplication. And if we're lucky it'll resolve that annoying double-tap we get from cmake.